dlpy/applications/efficientnet.py

```diff
@@ -0,0 +1,464 @@
+        if in_channels == out_channels and stride == 1 and (not noskip):
+            # dropout can be added.
+            return Res(name=prefix + 'add_se_residual')([x, inputs])
+        else:
+            return x
+
+    parameters = locals()
+    input_parameters = get_layer_options(input_layer_options, parameters)
+    inp = Input(**input_parameters, name='data')
+    # refer to Table 1  "EfficientNet-B0 baseline network" in paper:
+    # "EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks"
+    stage_id = 0
+    out_channels = round_filters(32, width_coefficient,
+                                 depth_divisor)  # multiply with width multiplier: width_coefficient
+    x = Conv2d(out_channels, 3, stride=2, include_bias=False, name='Conv1', act='identity')(inp)
+    x = BN(name='bn_Conv1', act=activation_fn)(x)
+
+    # Create stages with MBConv blocks from stage 1
+    in_channels = out_channels  # number of input channels for first MBblock
+    stage_id +=1
+    total_blocks = float(sum(args[2] for args in blocks_args))
+    for expansion, out_channels, num_blocks, ksize, stride, se_ratio in blocks_args:
+        out_channels = round_filters(out_channels, width_coefficient, depth_divisor)
+        num_blocks = round_repeats(num_blocks, depth_coefficient)
+        strides = [stride] + [1] * (num_blocks - 1)
+        for block_id, stride in enumerate(strides):
+            x = _MBConvBlock(x, in_channels, out_channels, ksize, stride, expansion, se_ratio, stage_id, block_id,activation_fn)
+            in_channels = out_channels  # out_channel
+        stage_id += 1
+
+    last_block_filters = round_filters(1280, width_coefficient, depth_divisor)
+    x = Conv2d(last_block_filters, 1, include_bias=False, name='Conv_top', act='identity')(x)
+    x = BN(name='Conv_top_bn', act=activation_fn)(x)
+
+    x = GlobalAveragePooling2D(name="Global_avg_pool", dropout=dropout_rate)(x)
+    x = OutputLayer(n=n_classes)(x)
+
+    model = Model(conn, inp, x, model_table)
+    model.compile()
+    return model
+
+
+def EfficientNetB0(conn, model_table='EfficientNetB0',
+                   n_classes=1000,
+                   **kwargs):
+    '''
+    Generates the EfficientNetB0 architecture.
+
+    Parameters
+    ----------
+    conn : CAS
+        Specifies the CAS connection object.
+    model_table : string or dict or CAS table, optional
+        Specifies the CAS table to store the deep learning model.
+    n_classes : int, optional
+        Specifies the number of classes. If None is assigned, the model will
+        automatically detect the number of classes based on the training set.
+        Default: 1000
+
+    References
+    ----------
+    https://arxiv.org/pdf/1905.11946.pdf
+    '''
+
+    return EfficientNet(conn, model_table, n_classes, n_channels=3,
+                        width=224, height=224, width_coefficient=1, depth_coefficient=1, dropout_rate=0.2,
+                        **kwargs)
```

Comment:
I dont find kwargs as a good fit for this content, I let it only for test purposes. Please update your code to reflect actual parameters

Reply:
fixed.